state/txindex/kv/kv.go

```diff
@@ -314,29 +315,30 @@ func (txi *TxIndex) match(
 
 	tmpHashes := make(map[string][]byte)
 
-	switch {
-	case c.Op == query.OpEqual:
+	switch c.Op {
+	case query.OpEqual:
 		it, err := dbm.IteratePrefix(txi.store, startKeyBz)
 		if err != nil {
 			panic(err)
 		}
 		defer it.Close()
 
+	EQ_LOOP:
 		for ; it.Valid(); it.Next() {
 			tmpHashes[string(it.Value())] = it.Value()
 
 			// Potentially exit early.
 			select {
 			case <-ctx.Done():
-				break
+				break EQ_LOOP
```

Comment:
Are we sure that, if the context passed as parameter to the function is "done" we just want to break out of the loop? (but still execute the one at the end: REMOVE_LOOP)

Reply:
Given the comment above the `select` statement I assumed that this is what was originally meant. Without specifying the `for` loop label, the existing code just breaks out of the `select` statement, which means we may as well just ignore the context altogether.

Reply:
The only alternative I can think of is to return from the function with an error.

What would trigger the termination of the context outside of the tests?

Reply:
Yes, I got the intention of you changes, and they are definitely improving the logic.
It's just that,  if the context is "done", don't we just want to return immediately? (if we're unsure, let's just leave it as you have it)

Reply:
Yeah I think this is something we need to clean up. I doubt it's catastrophic if we don't, but it is a bit messy right now.